### ISSUE TITLE
CUDA out of memory

### ISSUE THREAD

**Issue by sdimantsd** [dbolya/yolact#105]
Hi, sorry for all of my quastions and thanks so much for helping me!

When i try to eval an image, i get this error...
Im try it on this image:
https://f1.media.brightcove.com/8/1078702682/1078702682_6004950245001_6004956161001-vs.jpg?pubId=1078702682&videoId=6004956161001

If i trying it on "data/yolact_example_0.png" it's work fine

python3 eval.py --trained_model=yolact_im700_54_800000.pth --top_k=100 --image=car.jpg:asd.jpg
Config not specified. Parsed yolact_im700_config from the file name.

Loading model... Done.
Traceback (most recent call last):
  File "eval.py", line 1020, in <module>
    evaluate(net, dataset)
  File "eval.py", line 795, in evaluate
    evalimage(net, inp, out)
  File "eval.py", line 565, in evalimage
    img_numpy = prep_display(preds, frame, None, None, undo_transform=False)
  File "eval.py", line 199, in prep_display
    masks_color_cumul = masks_color[1:] * inv_alph_cumul
RuntimeError: CUDA out of memory. Tried to allocate 2.29 GiB (GPU 0; 7.93 GiB total capacity; 5.07 GiB already allocated; 1.57 GiB free; 832.34 MiB cached)

Im using 1080, 

$ nvidia-smi
Wed Jul 31 09:22:49 2019
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 410.48                 Driver Version: 410.48                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 1080    Off  | 00000000:01:00.0 Off |                  N/A |
| 20%   40C    P0    36W / 180W |      0MiB /  8117MiB |      5%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+

What i did wrong now????

**Comment by dbolya** [comment]
Looks like you forgot to set a score threshold. Your GPU doesn't have a lot of vram available (8gb), and you just asked it to generate 100 masks the size of your image, which despite its low quality, is 1920x1080. That's `100 * 1920 * 1080 * 4 * 3 = 2488320000` bytes or 2.48 gb of just raw mask visualization. It looks like there were a little bit less than 100 detections because it tried to allocate 2.29 gb in your error message, but looks like that's the culprit nonetheless.

I tend to like a top_k of 15 and a score threshold of 0.15, but you can tune these numbers to what works best for you. Try:
```
python3 eval.py --trained_model=yolact_im700_54_800000.pth --top_k=15 --score_threshold=0.15 --image=car.jpg:asd.jpg

**Comment by sdimantsd** [comment]
Yesssssss!!
Thanks!

**Comment by Zhang-O** [comment]
@dbolya， when I am in training, it seems that problems like "sdimantsd" occured with validating step. I do not find the place to set top_k in train.py.

Warning: Augmentation output an example with no ground truth. Resampling...
[  2]    2420 || B: 1.841 | C: 1.949 | M: 3.569 | S: 0.477 | T: 7.836 || ETA: 31 days, 18:18:29 || timer: 0.320
[  2]    2430 || B: 1.793 | C: 1.894 | M: 3.595 | S: 0.483 | T: 7.765 || ETA: 31 days, 16:33:04 || timer: 0.315
[  2]    2440 || B: 1.820 | C: 1.912 | M: 3.620 | S: 0.483 | T: 7.835 || ETA: 31 days, 19:18:32 || timer: 1.626
[  2]    2450 || B: 1.781 | C: 1.885 | M: 3.607 | S: 0.494 | T: 7.768 || ETA: 31 days, 18:13:23 || timer: 3.919

Computing validation mAP (this may take a while)...

Traceback (most recent call last):
  File "train.py", line 502, in <module>
    train()
  File "train.py", line 369, in train
    compute_validation_map(epoch, iteration, yolact_net, val_dataset, log if args.log else None)
  File "train.py", line 490, in compute_validation_map
    val_info = eval_script.evaluate(yolact_net, dataset, train_mode=True)
  File "/data/zhangyl/yolact-master/eval.py", line 955, in evaluate
    prep_metrics(ap_data, preds, img, gt, gt_masks, h, w, num_crowd, dataset.ids[image_idx], detections)
  File "/data/zhangyl/yolact-master/eval.py", line 403, in prep_metrics
    classes, scores, boxes, masks = postprocess(dets, w, h, crop_masks=args.crop, score_threshold=args.score_threshold)
  File "/data/zhangyl/yolact-master/layers/output_utils.py", line 91, in postprocess
    masks = F.interpolate(masks.unsqueeze(0), (h, w), mode=interpolation_mode, align_corners=False).squeeze(0)
  File "/home/ubuntu/anaconda3/envs/zyl/lib/python3.6/site-packages/torch/nn/functional.py", line 2503, in interpolate
    return torch._C._nn.upsample_bilinear2d(input, _output_size(2), align_corners)
RuntimeError: CUDA out of memory. Tried to allocate 4.84 GiB (GPU 0; 7.77 GiB total capacity; 695.44 MiB already allocated; 1.49 GiB free; 4.84 GiB cached)

**Comment by sdimantsd** [comment]
Try to train with smaller batch size.
Which GPU are you using?

**Comment by jdonovanCS** [comment]
@dbolya 

I'm getting a similar error as that of @Zhang-O . While computing validation mAP, I get a CUDA out of memory error.

Computing validation mAP (this may take a while)...

Traceback (most recent call last):
  File "train.py", line 504, in <module>
    train()
  File "train.py", line 371, in train
    compute_validation_map(epoch, iteration, yolact_net, val_dataset, log if args.log else None)
  File "train.py", line 492, in compute_validation_map
    val_info = eval_script.evaluate(yolact_net, dataset, train_mode=True)
  File "/media/jordan/LACIE/yolact/yolact/eval.py", line 956, in evaluate
    prep_metrics(ap_data, preds, img, gt, gt_masks, h, w, num_crowd, dataset.ids[image_idx], detections)
  File "/media/jordan/LACIE/yolact/yolact/eval.py", line 403, in prep_metrics
    classes, scores, boxes, masks = postprocess(dets, w, h, crop_masks=args.crop, score_threshold=args.score_threshold)
  File "/media/jordan/LACIE/yolact/yolact/layers/output_utils.py", line 91, in postprocess
    masks = F.interpolate(masks.unsqueeze(0), (h, w), mode=interpolation_mode, align_corners=False).squeeze(0)
  File "/home/jordan/anaconda3/envs/yolact-env/lib/python3.7/site-packages/torch/nn/functional.py", line 2530, in interpolate
    return torch._C._nn.upsample_bilinear2d(input, _output_size(2), align_corners)
RuntimeError: CUDA out of memory. Tried to allocate 6.68 GiB (GPU 0; 7.93 GiB total capacity; 742.42 MiB already allocated; 5.70 GiB free; 1.32 GiB reserved in total by PyTorch)

I am currently running with a batch_size=5 on a custom dataset. I just lowered it to 4 and will post back with results.
